invisible_cities/evm/pmaps_test.py

```diff
@@ -305,3 +315,58 @@ def test_PMap_s2s(pmps):
     assert len(pmp.s2s) == len(s2s)
     for kept_s2, true_s2 in zip(pmp.s2s, s2s):
         assert_Peak_equality(kept_s2, true_s2)
+
+
+
+@fixture(scope='module')
+def noise_func():
+    return NoiseSampler('new', 6400).signal_to_noise
+
+
+@fixture(scope='module')
+def s2_peak():
+    times      = np.arange(0, 20 * units.mus, 1 * units.mus)
+    bin_widths = np.full(len(times), 1 * units.mus)
+
+    pmt_ids  =    DB.DataPMT ('new', 6400).SensorID.values
+    n_sipms = len(DB.DataSiPM('new', 6400).SensorID.values)
+    sipm_ids = np.arange(n_sipms)
+
+    pmts  = PMTResponses(pmt_ids,
+                         np.random.uniform(0, 100,
+                                           (len(pmt_ids), len(times))))
+
+    sipms = SiPMResponses(sipm_ids,
+                          np.random.uniform(0, 10,
+                                            (len(sipm_ids), len(times))))
+
+    return S2(times, bin_widths, pmts, sipms)
+
+
+@mark.parametrize("charge_type",
+                  (SiPMCharge.raw, SiPMCharge.sn))
+def test_sipm_charge_array(charge_type,
+                           s2_peak    ,
+                           noise_func ):
+    charge_arr = s2_peak.sipm_charge_array(noise_func ,
+                                           charge_type,
+                                           False      )
+
+    all_wf = s2_peak.sipms.all_waveforms
+    assert np.array(charge_arr).shape[0] == all_wf.shape[1]
+    assert np.array(charge_arr).shape[1] == all_wf.shape[0]
+    assert np.count_nonzero(charge_arr) == np.count_nonzero(all_wf)
```

Comment:
Shouldn't they be exactly the same values? If so add an assertion for the values using `np.allclose`

Reply:
Again only in raw mode. Otherwise its s/N

Reply:
Mm ok. I think I would add some test that checks the output values. Is it too difficult?

Reply:
Looking into it

Reply:
I've put something together but I'm not sure I'm completely happy.

Let me know what you think

Reply:
Selecting a channel and a slice is because the array is too big, right? 20 number is not that much, but I guess it's ok. Besides, I would like those numbers to have more decimals to decrease the tolerance, but if it takes too long you can leave it as is. Finally, I would include the expected values thing inside the `s2_peak` fixture cause they are dependent on each other. Instead of 1 and 2, use something more descriptive like single_point or whatever you find appropriate (I am not very familiar with the nomenclature in this case)

Reply:
Will combine with S2_peak as you say.

On the shapes, 20 is the number of slices, there are 1792 channels so in the single case the returned tuple has shape (1792,) and for the array (20, 1792). Since I'm comparing to hand written values a slice seemed sufficient to check but...

Reply:
Yes, I agree on the channel, I was referring to the number of slices.

Reply:
wait, I thought we agreed on only one channel, why did you add more?

Reply:
I compare the 20 slices, 6 channels (one more than I had before).

Reply:
Ah! I think I misunderstood the shapes. I thought the 5 things (now 6) were the time slices... sorry about that. Anyway, I am not bothering you anymore. If you want to reduce the amount of numbers to check to make it less ugly feel free to do so.

Reply:
If this level of ugly is ok with the bosses, I'm happy not to touch it again!